File: CSharp/_VIPER Manutenção_/Code/Interfaces/VIPERInterfaces.cs
//
//  __MODULENAME__Interfaces.cs
//  __APPNAME__
//
//  Created by __USERNAME__ on __DATETIME__.
//  Copyright © __YEAR__ __ORGANIZATIONNAME__. All rights reserved.
//

using System.Windows.Forms;

namespace NAMESPACE.Interfaces
{
    public interface IViewToPresenterVIPER
    {
    }

    public interface IPresenterToViewVIPER
    {
    }

    public interface IPresenterToInteractorVIPER
    {
    }

    public interface IInteractorToPresenterVIPER
    {
    }

    public interface IPresenterToRouterVIPER
    {
        Form LoadModule();
    }
}

File: Delphi/$Padrão Manutenção Com Código Fonte$/Codigo/interfaces/VIPERInterfaces.pas
{
  __MODULENAME__Interfaces.pas
  __APPNAME__

  Created by __USERNAME__ on __DATETIME__.
  Copyright © __YEAR__ __ORGANIZATIONNAME__. All rights reserved.
}

unit VIPERInterfaces;

interface

uses
  System.Generics.Collections, Data.DB, Forms;

type
  IViewToPresenterVIPER = interface
  end;

  IPresenterToViewVIPER = interface
  end;

  IPresenterToInteractorVIPER = interface
  end;

  IInteractorToPresenterVIPER = interface
  end;

  IPresenterToRouterVIPER = interface
    procedure LoadModule(View: TForm);
  end;

implementation

end.
